sjsonnet/src/sjsonnet/Materializer.scala

```diff
@@ -29,7 +29,7 @@ object Materializer {
         val arrVisitor = visitor.visitArray(xs.length, -1)
         for(x <- xs) {
           arrVisitor.visitValue(
-            apply0(x.force, visitor),
```

Comment:
This was a bug in the existing implementations of `Materializer.scala`. The contract of `upickle.core.Visitor` is that you're meant to call `.subVisitor` and pass that in to recursive calls, to give the implementer a chance to hook in. This fixes `Materializer.scala` to be compliant